### ISSUE TITLE
Improve value type 'ALL' scan

### ISSUE THREAD

**Issue by Symbai** [cheat-engine/cheat-engine#1311]
If I search for anything higher than `255` I don't expect `BYTE` to be included in the search results but 2 bytes -> double. Same goes for `2 bytes` for anything higher than `65535` etc. Same also goes for rescan, i.e. when I search for `1` and then search for exact new value `123456789`, it should immediately remove all of these `byte` and `2 bytes` etc results but it currently does not. So every time there are a lot of false positive scan results.

**Comment by cheat-engine** [comment]
many people use the all type for cases they have no clue what type it is.  e.g.: If it's big endian searching for a value over 255 will not find the value they need, but it will find the correct address, it just won't be shown properly until it gets under 255 again

Even if not big endian, it's useful to scan for a value unrelated.  E.g civ6.  The 1 and 2 byte value all type can find the money as it's stored on an unaligned memory location and is only 3 bytes long

anyhow, in scan settings you can say that the 1 and 2 byte values are not part of the all scan

**Comment by Symbai** [comment]
> 
> 
> 
> anyhow, in scan settings you can say that the 1 and 2 byte values are not part of the all scan

No, it should do this on the fly based on the value I scan for. I use the ALL type a lot if I am not sure how the game stores the value. If I search for `1` it should include byte up to double. But if I search for `1000` it should NOT include byte results, still it does. The setting you mention I would need to change every time myself I perform a search, I dont think this is a solution to the issue.
